### ISSUE TITLE
Gradle 3.4.1 unable to build with FileNotFoundExcepiton(Operation not permitted)

### ISSUE THREAD

**Issue by cheehoo** [gradle/gradle#1565]
Hi All,

I been trying to clean build my gradle project with gradle wrapper 3.4.1 however was hitting the FileNotFoundException below

java.io.FileNotFoundException: /home/cheehoo/.gradle/caches/3.4.1/scripts-remapped/build_e2k02sjfohvdnii4nsaft7itr/c2d3oidpxg1t1bruezd0p3v75/proje7d8605dddf6ee8d441e8a198843b754/classes/build_e2k02sjfohvdnii4nsaft7itr$_run_closure11$_closure23$_closure25$_closure28$_closure29$_closure31.class (Operation not permitted )

I did try to find the file in the caches/3.4.1/script-remapped and it does not exist and i also tried to delete the cache folder however it still display the same errors. Anyone of encounter such errors before ?  

### Expected Behavior
Gradle should be able to build successfully without any errors

### Current Behavior
When building with gradle >=3.0(currently i am using 3.4.1 however i did tried few version and version 2.x is working fine) it present me with error FileNotFoundException as indicated above 

### Context
This issue have blocked me to build using gradle 3.4.1 

### Steps to Reproduce (for bugs)
./gradlew clean build

### Your Environment
CentOS release 6.7 on virtual box 4.3.20

------------------------------------------------------------
Gradle 3.4.1
------------------------------------------------------------

Build time:   2017-03-03 19:45:41 UTC
Revision:     9eb76efdd3d034dc506c719dac2955efb5ff9a93

Groovy:       2.4.7
Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015
JVM:          1.7.0_79 (Oracle Corporation 24.79-b02)
OS:           Linux 2.6.32-573.el6.x86_64 amd64

**Comment by lacasseio** [comment]
Thanks @cheehoo for the issue. Could you attach to the issue a reproducible example for us to play around with? Also, could you look into attaching a [build scan](https://gradle.com/scans/get-started) of your failing scenario?

**Comment by nvolk0616** [comment]
Just ran into this issue, and found that I needed to update my 'gradlew' file, as I had a 1.12 gradle version gradlew file.

My fix:

1. Installed the latest gradle on my machine
      a) Downloaded the latest gradle-3.*-bin.zip and extracted
      b) Set environment variable GRADLE_HOME to the extracted/installed directory
      c) Added GRADLE_HOME/bin to PATH
2. Switched to the base directory of my project where the existing gradlew file was located
3. Ran the newly installed gradle command to create new gradlew and gradlew.bat files ('gradle wrapper --gradle-version 3.4.1')

Was now able to use 'gradlew' on my project. Checked in the new gradlew and gradlew.bat files into version control.
